packages/gator-permissions-snap/src/clients/accountApiClient.ts

```diff
@@ -64,129 +72,213 @@ export type AccountApiClientConfig = {
+      1, // mainnet
+      10, // OP Mainnet
+      1329, // Sei Network
+      137, // Polygon Mainnet
+      42161, // Arbitrum One
+      534352, // Scroll Mainnet
+      56, // BSC Mainnet
+      59144, // Linea Mainnet
+      8453, // Base Mainnet
+    ].includes(chainId);
   }
 
   /**
-   * Fetch the token balance and metadata for a given account and token.
+   * Fetch the token balance for a given account and token. If the request fails, it will retry
+   * according to the retryOptions configuration.
    *
    * @param params - The parameters for fetching the token balance.
    * @param params.chainId - The chain ID to fetch the balance from.
    * @param params.assetAddress - The token address to fetch the balance for. If not provided, fetches native token balance.
    * @param params.account - The account address to fetch the balance for.
-   * @returns The token balance and metadata.
+   * @param params.retryOptions - Optional retry configuration. When not provided, defaults to 1 retry attempt with 1000ms delay.
+   * @returns The token balance as BigInt.
    */
-  public async getTokenBalanceAndMetadata({
+  public async getTokenBalance({
     chainId,
     assetAddress,
     account,
+    retryOptions,
   }: {
     chainId: number;
     account: Hex;
     assetAddress?: Hex | undefined;
-  }): Promise<TokenBalanceAndMetadata> {
+    retryOptions?: RetryOptions;
+  }): Promise<bigint> {
     if (!chainId) {
       const message = 'No chainId provided to fetch token balance';
       logger.error(message);
-
       throw new InvalidInputError(message);
     }
 
     if (!account) {
       const message = 'No account address provided to fetch token balance';
       logger.error(message);
-
       throw new InvalidInputError(message);
     }
-
     const tokenAddress = assetAddress ?? AccountApiClient.#nativeTokenAddress;
 
-    const validatedResponse = await makeValidatedRequest(
-      `${this.#baseUrl}/tokens/${tokenAddress}?accountAddresses=${account}&chainId=${chainId}`,
-      {
-        timeoutMs: this.#timeoutMs,
-        maxResponseSizeBytes: this.#maxResponseSizeBytes,
-        fetch: this.#fetch,
-      },
-      TokenBalanceResponseSchema,
+    // Try up to initial attempt + retry attempts
+    const response = await this.#fetchTokenBalance(
+      tokenAddress,
+      account,
+      chainId,
+      retryOptions,
     );
 
-    const { accounts, type, iconUrl, symbol, decimals } = validatedResponse;
+    // Parse and validate the response
+    const { balances, count } = response;
 
-    const accountLowercase = account.toLowerCase();
-    const accountData = accounts.find(
-      (acc) => acc.accountAddress.toLowerCase() === accountLowercase,
-    );
+    // If count is 0 or balances array is empty, it means the token has no balance (0 balance)
+    if (count === 0 || balances.length === 0) {
+      logger.info(
+        `No balance found for token ${tokenAddress} on account ${account}`,
+      );
+      return BigInt(0);
```

Comment:
nit: Prefer bigint literals.

```suggestion
      return 0n;
```